qcodes/dataset/experiment_container.py

```diff
@@ -218,6 +219,10 @@ def new_experiment(name: str,
         the new experiment
     """
     conn = conn or connect(get_DB_location())
+    exp_ids = get_matching_exp_ids(conn, name=name, sample_name=sample_name)
+    if len(exp_ids) >= 1:
+        log.warning(f"There is (are) already experiment(s) with the name of {name} "
+                    f"and sample name of {sample_name} in the database.")
```

Comment:
If we pass `sample_name=None` to the constructor of experiment it will use `some_sample` as the name so we really should check for that above ?

Reply:
Something like adding this to the function:
`sample_name = sample_name or "some_sample"`?

Reply:
Yes that should do the trick